packages/api/cms-api/src/page-tree/entities/page-tree-node-base.entity.ts

```diff
@@ -1,28 +1,26 @@
-import { BaseEntity, Entity, Enum, OptionalProps, PrimaryKey, Property } from "@mikro-orm/core";
+import { BaseEntity, Entity, Enum, Index, ManyToOne, OptionalProps, PrimaryKey, Property } from "@mikro-orm/core";
 import { Field, ID, Int, ObjectType } from "@nestjs/graphql";
 import { v4 as uuid } from "uuid";
 
-import { PageTreeNodeCategory, PageTreeNodeVisibility } from "../types";
+import { PageTreeNodeCategory, PageTreeNodeInterface, PageTreeNodeVisibility } from "../types";
 
 @Entity({ abstract: true })
 @ObjectType("PageTreeNodeBase", { isAbstract: true }) // ObjectType must be defined in base class! (The name "PageTreeNodeBase" is not used (we have no concrete type of PageTreeNodeBase))
-export abstract class PageTreeNodeBase extends BaseEntity<PageTreeNodeBase, "id"> {
+export abstract class PageTreeNodeBase extends BaseEntity {
     static tableName = "PageTreeNode";
     [OptionalProps]?: "createdAt" | "updatedAt" | "hideInMenu";
 
     @PrimaryKey({ columnType: "uuid" })
     @Field(() => ID)
     id: string = uuid();
 
-    @Property({ columnType: "uuid", nullable: true })
+    @Property({ columnType: "uuid", nullable: true, persist: false })
     @Field(() => String, { nullable: true })
     parentId: string | null;
 
-    // not possible in lib
-
-    // @ManyToOne(() => PageTreeNode)
-    // @Index()
-    // parent?: PageTreeNode;
+    @ManyToOne(() => "PageTreeNode", { nullable: true, joinColumn: "parentId" })
```

Comment:
Can we use the PAGE_TREE_ENTITY constant here? (like we do in the module to check the name)

https://github.com/vivid-planet/comet/blob/5b0f330616af5bd660bc82dc38b0f5698c3ff654/packages/api/cms-api/src/page-tree/page-tree.module.ts#L43-L45

Reply:
Yes: 154f0d1dce6a11126fa123e607431cf51ba4b39f